Assets/Scripts/Interaction/PlayerInteractionHandler.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using Unity.VisualScripting.YamlDotNet.Core.Tokens;
+using UnityEngine;
+
+namespace Cosmobot
+{
+    public class PlayerInteractionHandler : MonoBehaviour, DefaultInputActions.IInteractionActions
+    {
+        private IInteractable interaction = null;
+        private DefaultInputActions actions;
+        [SerializeField]
+        private TMP_Text interactionPrompt;
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.TryGetComponent(out IInteractable encounteredInteraction))
+            {
+                interaction = encounteredInteraction;
+                ShowInteractionPrompt();
+            }
+        }
+        private void OnTriggerExit(Collider other)
+        {
+            if (other.TryGetComponent(out IInteractable encounteredInteraction))
+            {
+                if (encounteredInteraction == interaction)
+                {
+                    interaction = null;
+                    interactionPrompt.enabled = false;
+                }
+            }
+        }
+        private void ShowInteractionPrompt()
+        {
+            interactionPrompt.text = interaction?.Prompt;
+            interactionPrompt.enabled = true;
+        }
+        private void Interact()
+        {
+            interaction?.Use();
+            ShowInteractionPrompt();
+        }
+        public void OnInteract(UnityEngine.InputSystem.InputAction.CallbackContext context)
+        {
+            if (context.performed)
+            {
+                Interact();
+            }
+        }
+        private void OnEnable()
+        {
+            if (actions == null)
```

Comment:
Can we init it at the start once?

Reply:
Well, yes, however not in Start fuction, because OnEnable is called before it. Awake is fine but i don't know if it would have any consequences later.

Reply:
keep it in `OnEnable`